rl4co/envs/scheduling/fjsp/env.py

```diff
@@ -0,0 +1,436 @@
+    then executed on the selected machine. The reward is 0 unless the agent scheduled all operations of all jobs.
+    In that case, the reward is (-)makespan of the schedule: maximizing the reward is equivalent to minimizing the makespan.
+
+    Observations:
+        - time: current time
+        - next_op: next operation per job
+        - proc_times: processing time of operation-machine pairs
+        - pad_mask: specifies padded operations
+        - start_op_per_job: id of first operation per job
+        - end_op_per_job: id of last operation per job
+        - start_times: start time of operation (defaults to 0 if not scheduled)
+        - finish_times: finish time of operation (defaults to INIT_FINISH if not scheduled)
+        - job_ops_adj: adjacency matrix specifying job-operation affiliation
+        - ops_job_map: same as above but using ids of jobs to indicate affiliation
+        - ops_sequence_order: specifies the order in which operations have to be processed
+        - ma_assignment: specifies which operation has been scheduled on which machine
+        - busy_until: specifies until when the machine will be busy
+        - num_eligible: number of machines that can process an operation
+        - job_in_process: whether job is currently being processed
+        - job_done: whether the job is done
+
+    Constrains:
+        the agent may not select:
+        - machines that are currently busy
+        - jobs that are done already
+        - jobs that are currently processed
+        - job-machine combinations, where the machine cannot process the next operation of the job
+
+    Finish condition:
+        - the agent has scheduled all operations of all jobs
+
+    Reward:
+        - the negative makespan of the final schedule
+
+    Args:
+        generator: FJSPGenerator instance as the data generator
+        generator_params: parameters for the generator
+        mask_no_ops: if True, agent may not select waiting operation (unless instance is done)
+    """
+
+    name = "fjsp"
+
+    def __init__(
+        self,
+        generator: FJSPGenerator = None,
+        generator_params: dict = {},
+        mask_no_ops: bool = True,
+        **kwargs,
+    ):
+        super().__init__(check_solution=False, **kwargs)
+        if generator is None:
+            if generator_params.get("file_path", None) is not None:
+                generator = FJSPFileGenerator(**generator_params)
+            else:
+                generator = FJSPGenerator(**generator_params)
+        self.generator = generator
+        self.num_mas = generator.num_mas
+        self.num_jobs = generator.num_jobs
+        self.n_ops_max = generator.max_ops_per_job * self.num_jobs
+        self.mask_no_ops = mask_no_ops
+        self._make_spec(self.generator)
+
+    def _set_seed(self, seed):
+        """Set the seed for the environment"""
+        rng = torch.manual_seed(seed)
+        self.rng = rng
```

Comment:
[Minor] this part is included in the `RL4COEnvBase()`: https://github.com/ai4co/rl4co/blob/684dbdcb6af83936a7733e814f1c9a42a280cab8/rl4co/envs/common/base.py#L267-L270